### PR TITLE
Improve parser typing

### PR DESCRIPTION
**Pull request by bartv**
# Description

I have been collecting typing improvements. This PR contains those related to the parser. It does a number of this:

1. It removes many unused methods from the stubs so that there are no useless warnings on the stubs anymore
2. It adds typing to those that we use
3. It already contains a number of fixes in the parser/compiler code
4. I added a bit of an ugly hack to correct type what comes out of the parser by making the stubs import Statement

There are quite some warnings left related to str vs locatable string but I am not familiar enough with this part of the code.

Anyone can contribute improvements if they have any but this is not high priority.

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] Attached issue to pull request
- [ ] Changelog entry
- [ ] Type annotations are present
- [ ] Code is clear and sufficiently documented
- [ ] No (preventable) type errors (check using make mypy or make mypy-diff)
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

# Reviewer Checklist:

- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Code is clear and sufficiently documented
- [ ] Correct, in line with design

### CODE REVIEW

File: src/inmanta/ast/statements/__init__.py [hunk 1]
```diff
@@ -208,9 +208,12 @@ def _add_to_dataflow_graph(self, graph: Optional[DataflowGraph]) -> None:
 
 
 class Literal(ExpressionStatement):
+    lexpos: Optional[int]
```
Comment (sanderr) [inline src/inmanta/ast/statements/__init__.py:211]: Why not declare the type in the constructor? I believe this technically declares it as a class attribute as well though in practice it makes little difference of course.

**Review by sanderr — APPROVED** [review]

File: src/inmanta/ast/statements/__init__.py (hunk 1 above)
Comment (bartv) [reply src/inmanta/ast/statements/__init__.py]: no idea. I fixed it.

Adding this is correct, no?

Comment (sanderr) [reply src/inmanta/ast/statements/__init__.py]: Adding what is correct?

Comment (bartv) [reply src/inmanta/ast/statements/__init__.py]: lexpos :)

Comment (sanderr) [reply src/inmanta/ast/statements/__init__.py]: Oh, for some reason I hadn't realized you'd added this, I thought you'd only added the typing. I'll double check.

**Review by wouterdb — APPROVED** [review]

File: src/inmanta/ast/statements/__init__.py (hunk 1 above)
Comment (sanderr) [reply src/inmanta/ast/statements/__init__.py]: Yeah, looks like it is set by the parser. I find it a bit strange but it certainly corresponds with the actual current type.

**Pull request closed without merging**